src/plugins/controls/public/control_types/options_list/options_list_embeddable.tsx

```diff
@@ -131,14 +131,16 @@ export class OptionsListEmbeddable extends Embeddable<OptionsListEmbeddableInput
 
     // push searchString changes into a debounced typeahead subject
     const typeaheadPipe = this.typeaheadSubject.pipe(
-      tap((newSearchString) => (this.searchString = newSearchString)),
-      debounceTime(100)
+      tap((newSearchString) => (this.searchString = newSearchString))
     );
 
     // fetch available options when input changes or when search string has changed
     this.subscriptions.add(
       merge(dataFetchPipe, typeaheadPipe)
-        .pipe(skip(1)) // Skip the first input update because options list query will be run by initialize.
+        .pipe(
+          skip(1), // Skip the first input update because options list query will be run by initialize.
+          debounceTime(100) // Skip all the updates emitted during the last n seconds, executing only the last one, if updates have stopped.
```

Comment:
Can you explain this change? Why did you move the debounce to the data fetch pipe? I believe this will still work alright, but my intention was to only have the typeahead fetch on a debounce, and to have query changes cause a re-fetch right away with no debounce.

Reply:
Here is a problem I was solving. These two pipelines were updating separately, one - for data change, the other - for typing at the search bar. There is a logic of canceling the queries (https://github.com/elastic/kibana/blob/8645a60405a1825a10804e0fa1db2d59bf05cabc/src/plugins/controls/public/control_types/options_list/options_list_embeddable.tsx#L239), located at the fetching method and on destroy. At the new flow, these two subjects are updating at the same time, typing with debounce, and both requests are canceling, so no results are received (first request is canceling on the second runOptionsListQuery call, and the second is canceled by `this.http.fetch(...)` internal logic, have no idea why (possibly, on destroy) (https://github.com/elastic/kibana/blob/8645a60405a1825a10804e0fa1db2d59bf05cabc/src/plugins/controls/public/services/kibana/options_list.ts#L67).
So, to avoid duplication of the requests I've decide to merge these subjects with debounce, what will prevent from running of two update requests at the same time.

WDTY?